docs/guides/fetch-data/01-read-profile-data.md

```diff
@@ -0,0 +1,520 @@
+          "hashFunction": "keccak256(bytes)",
+          "hash": "0x...",
+          "url": "..."
+        },
+        ...
+      ],
+      "tags": ["sample profile"]
+    }
+  },
+  "LSP1UniversalReceiverDelegate": "0x..."
+}
+```
+
+</details>
+
+## Step 3 - Read the Dataset
+
+With the JSON response, we can fetch all sorts of data:
+
+- Profile metadata
+- Pictures and their properties
+- Receiver address containing asset data
+
+Just instantiate variables for the information you would like to process. All demo fetch functions provided with this guide will come with a debug parameter. You can set it to `1` to see results directly in your terminal.
+
+<details>
+    <summary>Fetch the profile's metadata</summary>
+
+```javascript title="read_profile.js"
+...
+// Fetchable metadata information
+let name;
+let description;
+
+let links = [];
+let firstLinkTitle;
+let firstLinkURL;
+
+let tags = [];
+let firstTag;
+
+/*
+ * Fetch metadata information from the JSON dataset of
+ * an Universal Profile
+ */
+async function fetchProfileData(enableDebug) {
+  profileData = await getProfile(sampleProfileAddress);
+  name = profileData.LSP3Profile.LSP3Profile.name;
+  description = profileData.LSP3Profile.LSP3Profile.description;
+
+  links = profileData.LSP3Profile.LSP3Profile.links;
+  firstLinkTitle = links[0].title;
+  firstLinkURL = links[0].url;
+
+  tags = profileData.LSP3Profile.LSP3Profile.tags;
+  firstTag = tags[0];
+
+  if (enableDebug === 1) {
+    console.log('Name ' + name);
+    console.log('Description: ' + description + '\n');
+    console.log('Links: ' + JSON.stringify(links, undefined, 2) + '\n');
+    console.log('Title of first Link: ' + firstLinkTitle);
+    console.log('URL of first Link: ' + firstLinkURL + '\n');
+    console.log('Tags: ' + JSON.stringify(tags, undefined, 2) + '\n');
+    console.log('First tag: ' + firstTag + '\n');
+  }
+}
+
+// Debug
+fetchProfileData(1);
+```
+
+</details>
+
+<details>
+    <summary>Fetch the profile's picture properties</summary>
+
+```javascript title="read_profile.js"
+...
+// Fetchable picture information
+let baseURL = 'https://ipfs.lukso.network/ipfs/';
+
+let backgroundImageLinks = [];
+let fullSizeBackgroundImg;
+
+let profileImageLinks = [];
+let fullSizeProfileImg;
+
+/*
+ * Fetch picture information from the JSON dataset of
+ * an Universal Profile
+ */
+async function fetchPictureData(enableDebug) {
+  pictureData = await getProfile(sampleProfileAddress);
+  let backgroundImagesIPFS =
+    pictureData.LSP3Profile.LSP3Profile.backgroundImage;
+  let profileImagesIPFS = pictureData.LSP3Profile.LSP3Profile.profileImage;
+
+  for (let i in backgroundImagesIPFS) {
+    backgroundImageLinks.push([
+      i,
+      baseURL + backgroundImagesIPFS[i].url.substring(7),
+    ]);
+  }
+
+  for (let i in profileImagesIPFS) {
+    profileImageLinks.push([
+      i,
+      baseURL + profileImagesIPFS[i].url.substring(7),
+    ]);
+  }
+
+  fullSizeBackgroundImg = backgroundImageLinks[0][1];
```

Comment:
to do it correctly, you need to try/catch or handle case where there is no images.
The JSON documents may not be valid as they are set by anyone